### PR TITLE
WT-8055 Fix issue when compact quits when running at the same time as a checkpoint

### PR DESCRIPTION
**Pull request by AlexeyAnisimovAU**
This pull request implements the following changes:
1. Change all checkpoints in compact to "wait" checkpoints. This eliminates all conflicts between external (system/application) and internal compact checkpoints. If wait happens, additional checkpoint call shouldn't add too much overhead. Compact operation is expected to take a fairly long time. Adding a little extra delay shouldn't affect general performance too much.
2. "wt7989_compact_checkpoint" test now runs multiple checkpoints in parallel with compact. The first one starts immediately and the following checkpoints start after a random delay.

### CODE REVIEW

File: src/session/session_compact.c [hunk 1]
```diff
@@ -213,7 +233,7 @@ __compact_worker(WT_SESSION_IMPL *session)
     /*
      * Perform an initial checkpoint (see this file's leading comment for details).
      */
-    WT_ERR(__wt_session_blocking_checkpoint(session, true, session->compact->max_time));
+    WT_ERR(__compact_checkpoint(session));
```
Comment (keithbostic) [inline src/session/session_compact.c:236]: I think this means that `__w_session_blocking_checkpoint` can be static to its location, there's only a single caller now.

@kommiharibabu modeled his work there on top of how compact worked, so there also may be some changes to make there, that is, if compact was wrong (or at least, could be improved), maybe that code should be updated in the same way.

File: test/csuite/wt7989_compact_checkpoint/main.c [hunk 2]
```diff
@@ -166,12 +169,8 @@ run_test(bool stress_test, const char *home, const char *uri)
...
       file_sz_after / (1024.0 * 1024), file_sz_before / (1024.0 * 1024));
 
-    /*
-     * FIXME-WT-8055 At the moment the assert below is commented out to prevent evergreen from going
-     * red. Please enable the assert as soon as the underlying defect is fixed and compact does its
-     * job well.
-     */
-    /*testutil_assert(file_sz_before * 0.9 > file_sz_after);*/
+    /* Make sure the compact operation has reduced the file size by at least 20%. */
+    testutil_assert(file_sz_before * 0.8 > file_sz_after);
```
Comment (keithbostic) [inline test/csuite/wt7989_compact_checkpoint/main.c:173]: Using floating point operations on uint64_t seems a little weird to me, but it should work. OK, just getting over it... :)

File: test/csuite/wt7989_compact_checkpoint/main.c [hunk 3]
```diff
@@ -220,8 +220,13 @@ thread_func_checkpoint(void *arg)
 {
     struct thread_data *td;
     WT_SESSION *session;
+    time_t t;
+    uint64_t sleep_sec;
+    int i;
     bool signalled;
 
+    srand((u_int)time(&t));
```
Comment (keithbostic) [inline test/csuite/wt7989_compact_checkpoint/main.c:228]: `srand` is a bad RNG, but I'd agree we probably don't care much here, using srand/rand is fine.

**Review by keithbostic — APPROVED** [review]
lgtm

File: test/csuite/wt7989_compact_checkpoint/main.c (hunk 3 above)
Comment (sueloverso) [reply test/csuite/wt7989_compact_checkpoint/main.c:228]: All other tests, including many in `test/csuite` use `__wt_random`. This should too for consistency I think.

Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c]: Changed to `__wt_random_init_seed/__wt_random`. Thank you!

File: test/csuite/wt7989_compact_checkpoint/main.c (hunk 2 above)
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:173]: Changed to integer calculations: `(file_sz_before / 100) * 80 > file_sz_after`. Rounding error of up to 7920 bytes is acceptable giving the file size is 1378271232 bytes.

File: src/session/session_compact.c (hunk 1 above)
Comment (AlexeyAnisimovAU) [reply src/session/session_compact.c:236]: Good point. The only (probably not a viable) reason I left `__wt_session_blocking_checkpoint ` in public API is that there could be other, non-WT, consumers who use it. But I'm happy to make it static if you think it's OK.

Comment (kommiharibabu) [reply src/session/session_compact.c:236]: Yes, you can change it into a local function as it is used only in one place and also that function can be simplified as most of the options are not required as per the new code.

Comment (AlexeyAnisimovAU) [reply src/session/session_compact.c:236]: I've renamed `__wt_session_blocking_checkpoint` to `__session_blocking_checkpoint` and made it static. `force` and `seconds` parameters are not necessary anymore, so I removed them and the code associated with them. Thank you!

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Comment by AlexeyAnisimovAU** [comment]
Thank you for the review!

**Pull request merged**